docs/content/concepts/assets/asset-auto-execution.mdx

```diff
@@ -0,0 +1,169 @@
+
+1. Whether upstream data has changed
+2. Whether code has changed, which is tracked using [code versions](/concepts/assets/software-defined-assets#asset-code-versions)
+3. Any <PyObject object="FreshnessPolicy" pluralize /> set on the asset or downstream assets
+
+Assets can be auto-materialized "eagerly" – i.e. immediately after data or code changes occur. Or they can be auto-materialized "lazily" – i.e. by waiting until downstream <PyObject object="FreshnessPolicy" pluralize /> dictate that they need to be fresh. Or a mixture of both.
+
+Auto-materializing lazily is especially useful when dealing with source data that is constantly changing. As a batch system, Dagster
+
+<!-- TODO: some nice diagram that shows freshness policies making an impact backward and upstream changes making an impact forward? -->
+
+### Auto-materialize policies
+
+You can set up an asset to be auto-materialized by assigning it an <PyObject object="AutoMaterializePolicy" />. In this example, we use <PyObject object="AutoMaterializePolicy" method="eager" /> to indicate that, any time that `asset1` is materialized, `asset2` should be automatically materialized right after:
+
+```python file=/concepts/assets/auto_materialize_eager.py
+from dagster import AutoMaterializePolicy, asset
+
+
+@asset
+def asset1():
+    ...
+
+
+@asset(auto_materialize_policy=AutoMaterializePolicy.eager())
+def asset2(asset1):
+    ...
+```
+
+Note that the [Dagster Daemon](/some/link) needs to be running for auto-materialization to function. If you use the `dagster dev` command to run Dagster, the daemon will automatically run as part of it.
+
+### Auto-materialize policies and freshness policies
+
+Instead of auto-materializing downstream assets immediately after new upstream data arrives, you can use <PyObject object="AutoMaterializePolicy" method="lazy" /> to materialize assets only when needed to meet an asset's freshness policy. This allows avoiding unnecessary materializations.
+
+In this example, even if `asset1` is materialized every hour, `asset2` will only be materialized roughly once per day:
+
+```python file=/concepts/assets/auto_materialize_lazy.py
+from dagster import AutoMaterializePolicy, FreshnessPolicy, asset
+
+
+@asset
+def asset1():
+    ...
+
+
+@asset(
+    auto_materialize_policy=AutoMaterializePolicy.lazy(),
+    freshness_policy=FreshnessPolicy(maximum_lag_minutes=24 * 60),
+)
+def asset2(asset1):
+    ...
+```
+
+Freshness policies express how fresh an asset needs to be relative to data at the root of the graph. This means that, for an asset to meet it's freshness policy, both it and upstream assets need to be materialized in time.
+
+Setting a lazy auto-materialize policy on an asset allows it to be auto-materialized to help downstream assets meet their freshness policies. In this example, both `asset2` and `asset3` will be auto-materialized up to once per day, to help `asset3` meet its freshness policy.
```

Comment:
I think a natural question for a reader going through this guide is "do I need to add these policies to every asset?"

It might be worth making the counter point explicit here: what happens if asset 3 has a freshness policy but asset 1 or 2 does not have an auto materialization policy?

Reply:
> I think a natural question for a reader going through this guide is "do I need to add these policies to every asset?"

I'm going to add a section up here on how to add them to multiple assets at once.

I'll also incorporate your suggestion and make the counter-point more explicit.